components/views/MatesList/MatesList.tsx

```diff
@@ -1,15 +1,58 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { motion } from 'framer-motion'; // Import motion from framer-motion
-import useSWR from 'swr';
 import { fetcher } from '@/lib/fetcher';
 import { MateProps } from './Mate.interface';
 import { MateSkeleton } from './MateSkeleton';
 import { ErrorPage } from '@/components/Common/Error';
 import { Mate } from './Mate';
+import useSWRInfinite from 'swr/infinite';
 
 export const MatesList = () => {
-  const url = `/api/user/all`;
-  const { data, error, isLoading } = useSWR(url, fetcher);
+  const getKey = (pageIndex: number, previousPageData: MateProps[]) => {
+    let cursorId = '';
+    if (previousPageData) {
+      const lastMate = previousPageData[previousPageData.length - 1];
+      if (!lastMate) return null;
+      cursorId = lastMate.id;
+    }
+
+    if (pageIndex === 0) return `/api/user/all`;
+
+    return `/api/user/all?cursorId=${cursorId}&limit=15`;
+  };
+
+  const { data, size, setSize, error, isLoading } = useSWRInfinite<MateProps[]>(
+    getKey,
+    fetcher
+  );
+
+  const numberOfSKeletonPost = 15;
+  const skeletonMatesToLoad = Array.from({ length: numberOfSKeletonPost }, () =>
+    (Math.random() + 1).toString(36).substring(7)
+  );
+
+  const paginatedMates = data?.flat();
+
+  const isLoadingMore = data && typeof data[size - 1] === 'undefined';
+  const isNotReachEnd = data && data[data.length - 1].length;
+
+  const sortedPaginatedMates = (paginatedMates: MateProps[]) =>
```

Comment:
@rohitdasu  added sorting on UI level only instead db level, in this case the data that are currently visible on the screen will always be sorted, let me know if it will work for now.

Reply:
@csagar131 i don't believe it's a better solution - let's keep it without sorting. we'll create another issue for the task. thank you.

Reply:
@rohitdasu  sure, i have removed it.